### PR TITLE
chore(cli): Java init template compiles apps before trying to run them

### PR DESCRIPTION
**Pull request by ghost**
The TypeScript and C# init templates use commands that compile the code before running it to prevent the user from forgetting. This adds the same functionality to the Java init template.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by Jerry-AWS** [comment]
@RomainMuller Can you approve this, assuming I've done it the best way? I'd like to get it into a release soon; it'll let me remove lots of build instructions from the Dev Guide.

File: packages/aws-cdk/lib/init-templates/app/java/cdk.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  "app": "mvn -e -q exec:java"
+  "app": "mvn compile -q && mvn -e -q exec:java"
```
Comment (RomainMuller) [inline packages/aws-cdk/lib/init-templates/app/java/cdk.json:2]:
```suggestion
  "app": "mvn -e -q compile exec:java"
```

File: packages/aws-cdk/lib/init-templates/sample-app/java/cdk.json [hunk 2]
```diff
@@ -1,3 +1,3 @@
 {
-  "app": "mvn -e -q exec:java"
+  "app": "mvn compile -q && mvn -e -q exec:java"
```
Comment (RomainMuller) [inline packages/aws-cdk/lib/init-templates/sample-app/java/cdk.json:2]:
```suggestion
  "app": "mvn -e -q compile exec:java"
```

**Comment by RomainMuller** [comment]
> I'd like to get it into a release soon; it'll let me remove lots of build instructions from the Dev Guide.

Yeah that definitely is a sweet, sweet simplification! Thanks for looking into that!!

File: packages/aws-cdk/lib/init-templates/app/java/cdk.json (hunk 1 above)
Comment (Jerry-AWS) [reply packages/aws-cdk/lib/init-templates/app/java/cdk.json:2]: Figured there'd be a way to simplify it!

File: packages/aws-cdk/lib/init-templates/sample-app/java/cdk.json (hunk 2 above)
Comment (Jerry-AWS) [reply packages/aws-cdk/lib/init-templates/sample-app/java/cdk.json]: Done.

**Comment by Jerry-AWS** [comment]
@RomainMuller Ready for another look.

**Comment by Jerry-AWS** [comment]
> I reckon this can be achieved without shelling out twice (tested locally and achieved the expected result).

I thought about trying it in one command but was afraid it might still try to run the app if the compile failed. On reflection, even it did that, you're just going to get the same result you already had, so no harm. And avoiding spinning up the JVM twice seems like a definite win.

**Comment by shivlaks** [comment]
@Jerry-AWS nice! this is a great simplification.

were you able to run the integ tests successfully as well? `yarn integ-init` or `npm run integ-init`
since our PR build doesn't run these, might be worth running it for sanity ahead of the merge.

**Comment by RomainMuller** [comment]
Annotated with do-not-merge so you can consider @shivlaks' comment. You can remove the label when you're ready to merge in.

**Comment by Jerry-AWS** [comment]
I was able to run the integ tests successfully after a bit of effort.

**Pull request merged**